packages/flame_forge2d/lib/body_component.dart

```diff
@@ -74,36 +74,52 @@ abstract class BodyComponent<T extends Forge2DGame> extends Component
   @override
   void render(Canvas canvas) {
     if (renderBody) {
-      _renderFixtures(canvas);
+      body.fixtures.forEach(
+        (fixture) => renderFixture(canvas, fixture),
+      );
     }
   }
 
   @override
   void renderDebugMode(Canvas canvas) {
-    _renderFixtures(canvas);
+    body.fixtures.forEach(
+      (fixture) => renderFixture(canvas, fixture),
+    );
   }
 
-  void _renderFixtures(Canvas canvas) {
+  void _renderFixture(Canvas canvas, Fixture fixture) {
     canvas.save();
-    for (final fixture in body.fixtures) {
-      switch (fixture.type) {
-        case ShapeType.chain:
-          _renderChain(canvas, fixture);
-          break;
-        case ShapeType.circle:
-          _renderCircle(canvas, fixture);
-          break;
-        case ShapeType.edge:
-          _renderEdge(canvas, fixture);
-          break;
-        case ShapeType.polygon:
-          _renderPolygon(canvas, fixture);
-          break;
-      }
+    switch (fixture.type) {
+      case ShapeType.chain:
+        _renderChain(canvas, fixture);
+        break;
+      case ShapeType.circle:
+        _renderCircle(canvas, fixture);
+        break;
+      case ShapeType.edge:
+        _renderEdge(canvas, fixture);
+        break;
+      case ShapeType.polygon:
+        _renderPolygon(canvas, fixture);
+        break;
     }
     canvas.restore();
   }
 
+  /// Renders a [Fixture] in a [Canvas].
+  ///
+  /// Called for each fixture in [body] when [render]ing. Override this method
+  /// to customize how fixtures are rendered. For example, you can filter out
+  /// fixtures that you don't want to render.
+  ///
+  /// **NOTE**: If [renderBody] is false, no fixtures will be rendered. Hence,
+  /// [renderFixture] is not called when [render]ing.
+  void renderFixture(
+    Canvas canvas,
+    Fixture fixture,
+  ) =>
+      _renderFixture(canvas, fixture);
```

Comment:
Looks like the `_renderFixture()` method can be inlined here, merging the two.

Reply:
Indeed, that's how I had it initially. I think it fits well.

However, I went with `_renderFixture()` to keep with the file consistency, since `renderChain` has `_renderChain` and so forth. Those methods could be also a single method.

I'm in favour of not using `_renderFixture`. 

Wdyt @spydon ?

Reply:
Yeah, it doesn't add anything.